### PR TITLE
Issue 63: fixed issue 63 and added createdBy to popup table

### PR DESCRIPTION
**Pull request by erenfn**

### CODE REVIEW

File: backend/src/controllers/popupController.js [hunk 1]
```diff
@@ -37,7 +38,9 @@ class PopupController {
             }
         }
         try {
-            const newPopup = await popupService.createPopup(req.body);
+            console.log(userId)
+            const newPopupData = { ...req.body, createdBy: userId }; // Include userId in the data passed to the service
+            const newPopup = await popupService.createPopup(newPopupData);
             res.status(201).json(newPopup);
         } catch (err) {
             console.log(err);
```
Comment (uparkalau) [inline backend/src/controllers/popupController.js:46]: delete console log

File: backend/src/controllers/popupController.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ const Popup = require('../models/Popup');
 class PopupController {
 
     async addPopup(req, res) {
+        const userId = req.user.id;
         // Check if required fields are present and not null
```
Comment (uparkalau) [inline backend/src/controllers/popupController.js:11]: do not need this comments

File: backend/src/models/Popup.js [hunk 3]
```diff
@@ -1,12 +1,13 @@
 const { DataTypes } = require('sequelize');
 const sequelize = require('../../config/database');
+const User = require('./User'); // Adjust the path if necessary
```
Comment (uparkalau) [inline backend/src/models/Popup.js:3]: delete comments

**Review by uparkalau — CHANGES_REQUESTED** [review]
It looks like work in progress. you have to close this PR or finish code

**Comment by erenfn** [comment]
fixed

**Review by uparkalau — APPROVED** [review]

**Comment by uparkalau** [comment]
You can merge it

File: backend/src/controllers/popupController.js [hunk 4]
```diff
@@ -53,14 +48,12 @@ class PopupController {
         try {
             const { id } = req.params;
     
-            // Check if id is empty or not a number
             if (isNaN(id) || id.trim() === '') {
```
Comment (swoopertr) [inline backend/src/controllers/popupController.js:51]: @uparkalau we could have a validator function to check types for each field.

**Review by uparkalau — APPROVED** [review]
We can merge it into the develop

**Review by uparkalau — DISMISSED** [review]

**Review by uparkalau — APPROVED** [review]

**Pull request merged**